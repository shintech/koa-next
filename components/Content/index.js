import React from 'react'
import PropTypes from 'prop-types'
import Wrapper from './wrapper'
import SVG from 'static/images/react.svg'

const Content = ({ children, content = '' }) =>
  <Wrapper>
    {children}
    <SVG />
    <p>{ content }</p>
  </Wrapper>

Content.propTypes = {
  content: PropTypes.string.isRequired,
  children: PropTypes.node
}

export default Content
