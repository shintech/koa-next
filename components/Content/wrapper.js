import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  p {
    font-family: 'Farsan', cursive;
    font-size: 4ch;
    color: greenyellow;
    text-shadow: 0.18ch 0.18ch darkred;
    margin: 0;
  }

  svg {
    margin-bottom: 3ch;
    margin-top: 1.5ch;
    opacity: 0.7;
    width: 20ch;
    height: 20ch;
    :hover {
      opacity: 1;
    }
  }
`

export default Wrapper
