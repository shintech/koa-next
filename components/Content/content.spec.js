/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import Content from 'components/Content'

const shallowExpect = compose(expect, toJSON, shallow)

describe('COMPONENT -> Content...', () => {
  let clicker = shallow(<Content content='Hello World!' />)

  it(`expect Content p text to equal "Hello World!"...`, () => {
    expect(clicker.find('p').text()).toBe('Hello World!')
  })
})

describe('COMPONENT -> Content - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<Content content='Hello World!!' />).toMatchSnapshot()
  })
})
