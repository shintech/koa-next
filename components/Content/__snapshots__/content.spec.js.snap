// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Content - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper>
  <Component />
  <p>
    Hello World!!
  </p>
</wrapper__Wrapper>
`;
